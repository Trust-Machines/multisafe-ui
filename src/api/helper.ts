import {ClarityAbi} from '@stacks/transactions';
import {NETWORK} from '@trustmachines/multisafe-contracts';

export const validateSafeAbi = (abi: ClarityAbi) => {
    const publicFns = [
        'add-owner',
        'remove-owner',
        'set-threshold',
        'submit',
        'confirm',
        'revoke'
    ]

    for (let p of publicFns) {
        if (!abi.functions.find(x => x.name === p && x.access === 'public')) {
            return false;
        }
    }

    return true;
}

export const makeTxUrl = (txId: string, network: NETWORK) => {
    return `https://explorer.stacks.co/txid/${txId}?chain=${network}`;
}