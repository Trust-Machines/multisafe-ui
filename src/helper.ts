import {ClarityAbi} from '@stacks/transactions';
import {NETWORK} from '@trustmachines/multisafe-contracts';
import BigNumber from 'bignumber.js';

export const validateSafeAbi = (abi: ClarityAbi) => {
    const publicFns = [
        'add-owner',
        'remove-owner',
        'set-min-confirmation',
        'submit',
        'confirm',
        'revoke'
    ]

    for (let p of publicFns) {
        if (!abi.functions.find(x => x.name === p && x.access === 'public')) {
            return false;
        }
    }

    return true;
}

export const makeTxUrl = (txId: string, network: NETWORK) => {
    return `https://explorer.stacks.co/txid/${txId}?chain=${network}`;
}

export const formatUnits = (value: BigNumber | string, decimals: number): BigNumber => {
    const bn = typeof value === 'string' ? new BigNumber(value) : value;
    return bn.dividedBy(10 ** decimals);
}

export const parseUnits = (value: string, decimals: number): BigNumber => {
    return new BigNumber(value).multipliedBy(10 ** decimals);
}