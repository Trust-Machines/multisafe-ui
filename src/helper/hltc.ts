import { script, Network, networks, payments } from 'bitcoinjs-lib';
import {
    bytesToHex,
    IntegerType,
    intToHexString,
    hexToBytes,
    utf8ToBytes,
} from 'micro-stacks/common';
import { hashSha256 } from 'micro-stacks/crypto-sha';

export interface HTLC {
    hash: Uint8Array;
    senderPublicKey: Uint8Array;
    recipientPublicKey: Uint8Array;
    expiration?: number;
    swapper: IntegerType;
}

export type BufferType = Buffer | Uint8Array;

export function numberToLE(num: IntegerType, length = 4) {
    const hexBE = intToHexString(num, length);
    let le = '';
    // reverse the buffer
    for (let i = 0; i < length; i++) {
        le += hexBE.slice(-2 * (i + 1), -2 * i || length * 2);
    }
    return le;
}

export function numberToLEBytes(num: IntegerType, length = 4) {
    return hexToBytes(numberToLE(num, length));
}

export const CSV_DELAY = 500;
export const CSV_DELAY_BUFF = script.number.encode(CSV_DELAY);
export const CSV_DELAY_HEX = CSV_DELAY_BUFF.toString('hex');

export function encodeExpiration(expiration?: number): Buffer {
    return typeof expiration === 'undefined' ? CSV_DELAY_BUFF : script.number.encode(expiration);
}

export function htlcASM({ hash, senderPublicKey, recipientPublicKey, expiration, swapper }: HTLC) {
    const swapperHex = numberToLE(swapper);
    return `
  ${swapperHex} OP_DROP
	OP_IF
    OP_SHA256 ${bytesToHex(hash)}
    OP_EQUALVERIFY
		${bytesToHex(recipientPublicKey)}
	OP_ELSE
		${encodeExpiration(expiration).toString('hex')}
		OP_CHECKSEQUENCEVERIFY
		OP_DROP
		${bytesToHex(senderPublicKey)}
	OP_ENDIF
	OP_CHECKSIG`
        .replace(/\s+/g, ' ')
        .trim();
}

export function generateHTLCScript(htlc: HTLC) {
    const asm = htlcASM(htlc);
    const output = script.fromASM(asm);
    return output;
}

export function generateHTLCAddress(htlc: HTLC, _network?: Network) {
    const network = _network || networks.regtest;
    const script = generateHTLCScript(htlc);
    // const payment = payments.p2wsh({ redeem: { output: script, network }, network });
    // const payment = payments.p2sh({ , network });
    // const hash = ripemd160(Uint8Array.from(script));
    // const payment = payments.p2sh({ output: Buffer.from(hash), network });
    const payment = payments.p2sh({ redeem: { output: script, network }, network });
    return payment;
}

export function reverseBuffer(buffer: BufferType): Uint8Array {
    if (buffer.length < 1) return buffer;
    let j = buffer.length - 1;
    let tmp = 0;
    for (let i = 0; i < buffer.length / 2; i++) {
        tmp = buffer[i];
        buffer[i] = buffer[j];
        buffer[j] = tmp;
        j--;
    }
    return Uint8Array.from(buffer);
}

export function getScriptHash(output: BufferType): Uint8Array {
    const uintOutput = Uint8Array.from(output);
    const hash = hashSha256(uintOutput);
    const reversed = reverseBuffer(Buffer.from(hash));
    return reversed;
}

export function hexPadded(hex: string) {
    const bytes = hex.length % 2 ? `0${hex}` : hex;
    return bytes;
}

export function secretToHash(secret: string) {
    const bytes = utf8ToBytes(secret);
    return hashSha256(bytes);
}
