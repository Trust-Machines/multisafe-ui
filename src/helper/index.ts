import BigNumber from 'bignumber.js';
import {DEPLOYERS, NETWORK} from '@trustmachines/multisafe-contracts';
import {StacksNetwork} from '@stacks/network';
import {contractPrincipalCV, validateStacksAddress} from '@stacks/transactions';
import {NETWORKS} from '../constants';
import {escapeRegExp} from '../util';
import {TransactionType} from '../types';

export const formatUnits = (value: BigNumber | string, decimals: number): BigNumber => {
    const bn = typeof value === 'string' ? new BigNumber(value) : value;
    return bn.dividedBy(10 ** decimals);
}

export const parseUnits = (value: string, decimals: number): BigNumber => {
    return new BigNumber(value).multipliedBy(10 ** decimals);
}

export const checkDecimalAmount = (input: string) => {
    const re = RegExp(`^\\d*(?:\\\\[.])?\\d*$`);

    if (!re.test(escapeRegExp(input))) {
        return false;
    }

    if (input.indexOf('.') !== -1) {
        return input.split('.')[1].length <= 6;
    }

    return true;
}

export const checkAmount = (input: string) => {
    return RegExp(/^(\s*|\d+)$/).test(input)
}

export const getStacksNetwork = (n: NETWORK): StacksNetwork => {
    return NETWORKS[n];
}

export const detectTransactionType = (executor: string): TransactionType => {
    const exec = executor.replace("'", '');
    for (const d of DEPLOYERS) {
        if (`${d}.add-owner` === exec) {
            return 'add-owner';
        }

        if (`${d}.remove-owner` === exec) {
            return 'remove-owner';
        }

        if (`${d}.set-threshold` === exec) {
            return 'set-threshold';
        }

        if (`${d}.transfer-stx` === exec) {
            return 'transfer-stx';
        }

        if (`${d}.transfer-sip-009` === exec) {
            return 'transfer-sip-009';
        }

        if (`${d}.transfer-sip-010` === exec) {
            return 'transfer-sip-010';
        }
    }

    return null;
}

export const contractPrincipalCVFromString = (s: string) => {
    const [a, b] = s.split(".");
    return contractPrincipalCV(a, b);
}

export const transformNftUri = (uri: string, nftId: string) => {
    const u = uri.replace('{id}', nftId);

    if (u.startsWith('ipfs://')) {
        return u.replace(/ipfs:\/\/(ipfs\/)?/, 'https://gateway.ipfs.io/ipfs/');
    }

    if (u.startsWith('ar://')) {
        return u.replace('ar://', 'https://arweave.net/');
    }

    return u;
}

export const isValidRecipient = (r: string) => {
    if (r === '') {
        return false;
    }

    if (validateStacksAddress(r)) {
        return true;
    }

    try {
        contractPrincipalCVFromString(r);
        return true;
    } catch (e) {

    }

    return false;
}
