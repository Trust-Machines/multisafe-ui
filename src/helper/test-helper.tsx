import {render} from '@testing-library/react';
import {createHistory, createMemorySource, LocationProvider} from '@reach/router';

export const renderWithRouter = (
    ui: any,
    {route = '/', history = createHistory(createMemorySource(route))} = {},
) => {
    return {
        ...render(<LocationProvider history={history}>{ui}</LocationProvider>),
        // adding `history` to the returned utilities to allow us
        // to reference it in our tests (just try to avoid using
        // this to test implementation details).
        history,
    }
}