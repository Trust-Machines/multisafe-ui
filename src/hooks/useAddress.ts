import useUserSession from './useUserSession';
import useNetwork from './useNetwork';

const useAddress = (): string | null => {
    const [, userData] = useUserSession();
    const [network] = useNetwork();

    if (userData) {
        return userData.profile.stxAddress[network];
    }

    return null;
}

export default useAddress;