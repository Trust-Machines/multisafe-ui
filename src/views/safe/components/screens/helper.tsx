import React from 'react';
import Typography from '@mui/material/Typography';
import {Box, useTheme} from '@mui/material';

const ScreenHeader = (props: { title: string, icon: JSX.Element, children: JSX.Element }) => {
    const theme = useTheme();
    const icon = React.cloneElement(props.icon, {
        sx: {
            marginRight: '4px',
            color: theme.palette.primary.main
        }
    })
    return <Box sx={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between',
        mb: '20px'
    }}>
        <Typography variant="h6" sx={{display: 'flex', alignItems: 'center',}}>
            {icon} {props.title}
        </Typography>
        {props.children}
    </Box>
}

export default ScreenHeader;