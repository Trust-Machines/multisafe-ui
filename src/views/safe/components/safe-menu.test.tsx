import SafeMenu from './safe-menu';
import {renderWithRouter} from '../../../helper/test-helper';
import useSafe from '../../../hooks/use-safe';
import useAddress from '../../../hooks/use-address';

jest.mock('../../../hooks/use-safe');
jest.mock('../../../hooks/use-address');

test('1 Render', () => {
    (useSafe as jest.Mock).mockReturnValue([{fullAddress: 'SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7.my-safe'}])
    const rendered = renderWithRouter(<SafeMenu section="nft"/>);
    expect(rendered.container).toMatchSnapshot()
});

test('2 Render with owner', () => {
    (useSafe as jest.Mock).mockReturnValue([{
        fullAddress: 'SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7.my-safe',
        owners: [
            'SP2N7SK0W83NJSZHFH8HH31ZT3DXJG7NFE5VYT9SJ',
            'SP2DXHX9Q844EBT80DYJXFWXJKCJ5FFAX50CQQAWN',
            'SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7'
        ]
    }]);

    (useAddress as jest.Mock).mockReturnValue('SP2DXHX9Q844EBT80DYJXFWXJKCJ5FFAX50CQQAWN')

    const rendered = renderWithRouter(<SafeMenu section="nft"/>);
    expect(rendered.container).toMatchSnapshot()
});

