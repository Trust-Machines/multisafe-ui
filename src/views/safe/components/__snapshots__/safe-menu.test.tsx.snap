// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 Render 1`] = `
<div>
  <div
    class="MuiBox-root css-nu4g12"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeLarge css-p4wudk-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
  <div
    class="MuiBox-root css-6egkrn"
  >
    <div
      class="MuiBox-root css-pzu2px"
    >
      <div
        class="MuiBox-root css-14iyp89"
      />
      <div
        class="MuiBox-root css-1w2dia5"
      >
        <img
          alt="SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7.my-safe"
          src="data:image/png;base64,iVBORw0..."
          style="border-radius: 50%;"
        />
      </div>
      <div
        class="MuiBox-root css-1t0990z"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SP3XD84X....my-safe
        </p>
      </div>
      <div
        class="MuiBox-root css-urcpk1"
      >
        <svg
          aria-hidden="true"
          aria-label="Copied"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-6x0rx9-MuiSvgIcon-root"
          data-mui-internal-clone-element="true"
          data-testid="ContentCopyIcon"
          focusable="false"
          style="cursor: pointer;"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
          />
        </svg>
        <a
          class="MuiBox-root css-70qvj9"
          href="https://explorer.stacks.co/txid/SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7.my-safe?chain=mainnet"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-121cfi1-MuiSvgIcon-root"
            data-testid="OpenInNewIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </a>
        <div
          class="MuiBox-root css-759u60"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hvr4hn-MuiSvgIcon-root"
            data-testid="RefreshIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiBox-root css-wdzsr9"
      >
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorInfo MuiChip-outlinedInfo css-rofncn-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            Readonly
          </span>
        </div>
        
      </div>
    </div>
    <nav
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="TollIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 4c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zM3 12c0-2.61 1.67-4.83 4-5.65V4.26C3.55 5.15 1 8.27 1 12s2.55 6.85 6 7.74v-2.09c-2.33-.82-4-3.04-4-5.65z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Coins
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters Mui-selected css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DiamondIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.16 3h-.32L9.21 8.25h5.58zm4.3 5.25h5.16L19 3h-5.16zm4.92 1.5h-8.63V20.1zM11.25 20.1V9.75H2.62zM7.54 8.25 10.16 3H5L2.38 8.25z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            NFTs
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SwapVertIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 17.01V10h-2v7.01h-3L15 21l4-3.99h-3zM9 3 5 6.99h3V14h2V6.99h3L9 3z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Transactions
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="GroupsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12.75c1.63 0 3.07.39 4.24.9 1.08.48 1.76 1.56 1.76 2.73V18H6v-1.61c0-1.18.68-2.26 1.76-2.73 1.17-.52 2.61-.91 4.24-.91zM4 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm1.13 1.1c-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58C.48 14.9 0 15.62 0 16.43V18h4.5v-1.61c0-.83.23-1.61.63-2.29zM20 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85-.85-.37-1.79-.58-2.78-.58-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V18H24v-1.57zM12 6c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Owners
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FactCheckIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM10 17H5v-2h5v2zm0-4H5v-2h5v2zm0-4H5V7h5v2zm4.82 6L12 12.16l1.41-1.41 1.41 1.42L17.99 9l1.42 1.42L14.82 15z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Policy
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </nav>
  </div>
</div>
`;

exports[`2 Render with owner 1`] = `
<div>
  <div
    class="MuiBox-root css-nu4g12"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeLarge css-p4wudk-MuiSvgIcon-root"
      data-testid="MenuIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
  <div
    class="MuiBox-root css-6egkrn"
  >
    <div
      class="MuiBox-root css-pzu2px"
    >
      <div
        class="MuiBox-root css-14iyp89"
      />
      <div
        class="MuiBox-root css-1w2dia5"
      >
        <img
          alt="SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7.my-safe"
          src="data:image/png;base64,iVBORw0..."
          style="border-radius: 50%;"
        />
      </div>
      <div
        class="MuiBox-root css-1t0990z"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          SP3XD84X....my-safe
        </p>
      </div>
      <div
        class="MuiBox-root css-urcpk1"
      >
        <svg
          aria-hidden="true"
          aria-label="Copied"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-6x0rx9-MuiSvgIcon-root"
          data-mui-internal-clone-element="true"
          data-testid="ContentCopyIcon"
          focusable="false"
          style="cursor: pointer;"
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"
          />
        </svg>
        <a
          class="MuiBox-root css-70qvj9"
          href="https://explorer.stacks.co/txid/SP3XD84X3PE79SHJAZCDW1V5E9EA8JSKRBPEKAEK7.my-safe?chain=mainnet"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-121cfi1-MuiSvgIcon-root"
            data-testid="OpenInNewIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </a>
        <div
          class="MuiBox-root css-759u60"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hvr4hn-MuiSvgIcon-root"
            data-testid="RefreshIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiBox-root css-wdzsr9"
      >
        
        <div
          class="MuiChip-root MuiChip-outlined MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-outlinedSuccess css-c8195h-MuiChip-root"
        >
          <span
            class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            Owner 2
          </span>
        </div>
      </div>
    </div>
    <nav
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="TollIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 4c-4.42 0-8 3.58-8 8s3.58 8 8 8 8-3.58 8-8-3.58-8-8-8zm0 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6zM3 12c0-2.61 1.67-4.83 4-5.65V4.26C3.55 5.15 1 8.27 1 12s2.55 6.85 6 7.74v-2.09c-2.33-.82-4-3.04-4-5.65z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Coins
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters Mui-selected css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="DiamondIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.16 3h-.32L9.21 8.25h5.58zm4.3 5.25h5.16L19 3h-5.16zm4.92 1.5h-8.63V20.1zM11.25 20.1V9.75H2.62zM7.54 8.25 10.16 3H5L2.38 8.25z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            NFTs
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SwapVertIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 17.01V10h-2v7.01h-3L15 21l4-3.99h-3zM9 3 5 6.99h3V14h2V6.99h3L9 3z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Transactions
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="GroupsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12.75c1.63 0 3.07.39 4.24.9 1.08.48 1.76 1.56 1.76 2.73V18H6v-1.61c0-1.18.68-2.26 1.76-2.73 1.17-.52 2.61-.91 4.24-.91zM4 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm1.13 1.1c-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58C.48 14.9 0 15.62 0 16.43V18h4.5v-1.61c0-.83.23-1.61.63-2.29zM20 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85-.85-.37-1.79-.58-2.78-.58-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V18H24v-1.57zM12 6c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3z"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Owners
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
      <div
        class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters css-16ac5r2-MuiButtonBase-root-MuiListItemButton-root"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FactCheckIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM10 17H5v-2h5v2zm0-4H5v-2h5v2zm0-4H5V7h5v2zm4.82 6L12 12.16l1.41-1.41 1.41 1.42L17.99 9l1.42 1.42L14.82 15z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Policy
          </span>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </div>
    </nav>
  </div>
</div>
`;
